dspace-api/src/main/java/org/dspace/identifier/doi/DOIOrganiser.java

```diff
@@ -799,4 +661,81 @@ private void setQuiet() {
         this.quiet = true;
     }
 
-}
\ No newline at end of file
+    private void batchProcess(List<DOI> doiList, DOIOperation operation, String operationName) {
+        int batchSize = getBatchSize();
+        int count = 0;
+
+        try {
+            for (DOI doiRow : doiList) {
+                DSpaceObject dso = doiRow.getDSpaceObject();
+                if (Constants.ITEM != dso.getType()) {
+                    LOG.warn("Skipping DOI " + doiRow.getDoi() + ": Currently DSpace supports DOIs for Items only.");
+                    continue;
+                }
+
+                try {
+                    // Reload the DSpaceObject to ensure it's attached to the current session
+                    dso = context.reloadEntity(dso);
+                    doiRow.setDSpaceObject(dso);  // Update the DOI object with the reloaded DSpaceObject
+
+                    operation.perform(context, provider, doiRow);
+                    count++;
+
+                    if (count % batchSize == 0) {
+                        context.commit();
+                        if (!quiet) {
+                            System.out.println("Processed " + count + " DOIs for " + operationName + ".");
+                        }
+                    }
+
+                    if (!quiet) {
+                        System.out.println("Successfully " + operationName + " DOI " + DOI.SCHEME + doiRow.getDoi() + ".");
+                    }
+                } catch (IdentifierException | SQLException ex) {
+                    LOG.error("Error " + operationName + " DOI " + doiRow.getDoi(), ex);
+                }
+
+                if (count % batchSize == 0) {
+                    context.uncacheEntity(doiRow);
+                }
```

Comment:
We are removing from the cache only the last processed doi of each batch.
It should be applied to all `doiRow`s.